scr/main/java/bittercode/util/DatabaseConfig.java

```diff
@@ -0,0 +1,43 @@
+package com.bittercode.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+class DatabaseConfig {
+
+    static Properties prop = new Properties();
+    static {
+
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        InputStream input = classLoader.getResourceAsStream("application.properties");
+
+        try {
+            prop.load(input);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    try  {
+        input.close();
+    } catch (IOException e) {
+        e.printStackTrace();
+    }
+
+    try   {
+        input.close();
+    } catch (IOException e) {
+        e.printStackTrace();
+    }

```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider removing duplicate stream close blocks</b></div>
<div id="fix">

Consider removing the duplicate `input.close()` blocks. The stream is being closed twice which is redundant and could potentially cause issues.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -15,20 +15,10 @@
          try {
              prop.load(input);
 +            input.close();
          } catch (IOException e) {
              e.printStackTrace();
          }
      }
 -
 -    try  {
 -        input.close();
 -    } catch (IOException e) {
 -        e.printStackTrace();
 -    }
 -
 -    try   {
 -        input.close();
 -    } catch (IOException e) {
 -        e.printStackTrace();
 -    }

```


```suggestion
        try {
            prop.load(input);
            input.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/7#issuecomment-2527741727>#f9242c</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged